### PR TITLE
Interact with DICOM Operations Storage from DICOM Server

### PR DESCRIPTION
**Pull request by wsugarman**
## Description
This PR removes the HTTP triggers and embeds their logic within the DICOM server itself.
- Remove Azure Function HTTP triggers
- Update `GetStatusAsync` and `StartQueryTagIndexingAsync` to use an `IDurableClient`, added through DI, that communicates directly with the function storage
- Refactor/move code around to accommodate logic now residing in the `Microsoft.Health.Dicom.Functions.Client` project, including unit tests
- Update app.config to ensure F5 experience still works
- Update dockerfile and docker-compose project as the function key is no longer necessary
- Update ARM template to no longer store function key
- Update function-related packages and Azurite image
- Alphabetize configs

## Related issues
TBD

## Testing
Locally and via pipeline

### CODE REVIEW

**Comment by wsugarman** [comment]
Still pending:
- Review files for obvious style, comment, or namespace issues
- Prune NuGet dependencies
- Potentially rename projects

File: src/Microsoft.Health.Dicom.Functions.Client/Extensions/DurableOrchestrationStatusExtensions.cs [hunk 1]
```diff
@@ -7,20 +7,17 @@
...
         public static OperationType GetOperationType(this DurableOrchestrationStatus durableOrchestrationStatus)
         {
             EnsureArg.IsNotNull(durableOrchestrationStatus, nameof(durableOrchestrationStatus));
 
-            return string.Equals(
-                    durableOrchestrationStatus.Name,
-                    nameof(ReindexDurableFunction.ReindexInstancesAsync),
-                    StringComparison.OrdinalIgnoreCase)
+            return durableOrchestrationStatus.Name != null &&
+                durableOrchestrationStatus.Name.StartsWith(FunctionNames.ReindexInstances, StringComparison.OrdinalIgnoreCase)
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Functions.Client/Extensions/DurableOrchestrationStatusExtensions.cs:20]: In the future, we may have multiple versions of the same function.

File: src/Microsoft.Health.Dicom.Web/appsettings.Development.json [hunk 2]
```diff
@@ -1,17 +1,23 @@
 {
+  "BlobStore": {
+    "ConnectionString": "UseDevelopmentStorage=true"
+  },
+  "DicomFunctions": {
+    "ConnectionName": "BlobStore:ConnectionString"
```
Comment (pengchen0692) [inline src/Microsoft.Health.Dicom.Web/appsettings.Development.json:6]: Should `ConnectionName` be in appsettings.json?

File: src/Microsoft.Health.Dicom.Functions.Client/DicomAzureFunctionsClient.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        }
+
+        /// <inheritdoc/>
+        public async Task<Guid> StartQueryTagIndexingAsync(IReadOnlyCollection<int> tagKeys, CancellationToken cancellationToken = default)
+        {
+            EnsureArg.IsNotNull(tagKeys, nameof(tagKeys));
+            EnsureArg.HasItems(tagKeys, nameof(tagKeys));
+
+            // Start the re-indexing orchestration
+            // TODO: Pass token when supported
```
Comment (pengchen0692) [inline src/Microsoft.Health.Dicom.Functions.Client/DicomAzureFunctionsClient.cs:109]: nit: move `TODO` next to `_durableClient.StartNewAsync`

**Review by pengchen0692 — APPROVED** [review]

File: src/Microsoft.Health.Dicom.Web/appsettings.Development.json (hunk 2 above)
Comment (wsugarman) [reply src/Microsoft.Health.Dicom.Web/appsettings.Development.json:6]: It is pretty specific to running locally, so I'm reluctant to put it outside of the development settings

Comment (wsugarman) [reply src/Microsoft.Health.Dicom.Web/appsettings.Development.json:6]: But I have changed the format of the config to better illustrate it isn't necessarily the same as blob storage

**Pull request merged**